README.md

```diff
@@ -117,6 +124,38 @@ For testing purposes you can create your own LoaderFactory. For ease of use, Str
 
 3rd, 4th and 5th step are same as in previous example.
 
+#### Writing tests using minSdk value
+
+All the steps are the same just like writing tests without minSdk values. The only and single difference in writing tests with minSdk values is the PrinceOfVersions object, to be more precise, it's constructor's arguments. 
+
+```
+PrinceOfVersions updater = new PrinceOfVersions(context, provider, repository, setupSdkInt(20));
+```
+
+Since we've added the support for minSdk values of the device you can mock and customize them when writing tests by using the method <code>setupSdkInt(int minSdkValue)</code>.
+
+When creating a PrinceOfVersions object a few things need to be kept in mind:
+
+* <code>context</code> argument in PrinceOfVersions constructor can be made using the <code>setupContext(String versionString)</code> where versionString is a variation of one of the following strings: 
+
+<code>"1.0.0"</code> <code>"1.2.3"</code> <code>"1.0.1-b12"</code> <code>"1.0.1-rc3"</code> <code>"1.0.1-beta4"</code>
+
+* <code>provider</code> argument in PrinceOfVersions constructor can be Mocked using Mockito library. 
+
+```
+provider = Mockito.mock(VersionVerifierFactory.class);
+```
+
+and its used for creating a new instance of specific <code>VersionVerifier</code> and it has a single method that provides a new instance of <code>VersionVerifier</code> which is used for verifying updates and cancellation of verification.
+
+* <code>repository</code> argument is used for representing repository which persists library data and is also mocked with Mockito library. 
+
+```
+repository = Mockito.mock(VersionRepository.class);
+```
+
+* And finally, <code>setupSdkInt(int minSdkValue)</code> this method mocks the minSdk value and treats and makes the lib act as if user is using a device with <code>minSdkValue</code> you passed as an argument.
```

Comment:
I still do not understand the _setupSdkInt_ method. That is a method you created in library tests, but a normal user can not use it because it does not exist in the API.